node/pkg/common/queryResponse.go

```diff
@@ -29,6 +42,13 @@ type QueryResponsePublication struct {
 	Response EthCallQueryResponse
 }
 
+func (resp *QueryResponsePublication) RequestID() string {
+	if resp == nil || resp.Request == nil {
+		return "nil"
+	}
+	return hex.EncodeToString(resp.Request.Signature)
```

Comment:
It's possible we should make this the hash of the request (body + sig for offchain), but this should work for now. Once we add on-chain request support, we will likely need to change the request to a struct anyway. :thinking: Actually wondering if it should be the hash of that first part of the serialized request (chainId + sig or vaa hash)

Reply:
Made a note to rethink the request ID.